File: Algo_Method/multiple_algo/Q2-7/Q2-7.rb
def f(x) return x * (x + 1) / 2
end

n = gets.chomp.to_i
gets.chomp.split.map(&:to_i).each do |val|
    l = -1
    r = 2 * 10 ** 9
    while l != r do
        mid = (l + r) / 2
        if f(mid) >= val
            r = mid
        else 
            l = mid + 1
        end
    end
    p l
end

File: Algo_Method/multiple_algo/Q2-7/Q2-7.py
def f(x): return x * (x + 1) // 2
n = int(input())
for val in list(map(int, input().split())):
    l, r = -1, 2 * 10 ** 9
    while l != r:
        mid = (l + r) // 2
        if f(mid) >= val: r = mid
        else: l = mid + 1
    print(l)